packages/nimble-components/src/table/specs/data-hierarchy-hld.md

```diff
@@ -0,0 +1,199 @@
+    }
+
+    // The set of rows the user would like to collapse. Pass 'true' for `collapseChildren` if all
+    // children rows parented under any specified in rowIds should also be collapsed.
+    public async collapseRows(rowIds: string[], collapseChildren?: boolean) Promise<void> {
+        ...
+    }
+}
+```
+
+### `TableRowExpandedEventDetail` API
+
+The `Table` will also provide a `row-expand-toggle` event for when a row is expanded/collapsed that will provide details to the client including which row was expanded/collapsed, and what its new state is:
+
+```ts
+interface TableRowExpandedEventDetail {
+    newState: boolean;
+    oldState: boolean;
+    recordId: string;
+}
+```
+
+Note: This event will _not_ be emitted for group rows.
+
+### Interactions
+
+The following are various expected mouse and keyboard interactions related to parent rows:
+
+1. The collapse-all button in the header will also collapse parent rows.
+2. Clicking on a parent row (not on the expand/collapse button) will select the parent row. If the table `selectionMode` is set to `none`, then if a user clicks on a parent row (again _not_ on the expand/collapse button) nothing occurs.
+3. Keyboard interactions are being defined as part of the [Table Keyboard Navigation HLD](https://github.com/ni/nimble/pull/1506).
+
+### Validation
+
+Tha table will be invalid if the user has set the `parentIdFieldName` attribute but not the `idFieldName`. Additionally, the table will be invalid if the user has set the `forceExpandCollapseFieldName` attribute but not the `parentIdFieldName`.
+
+### Lazy Loading
+
+The APIs noted above will enable the client to lazy load data into the `Table`. This will essentially be accomplished with the following steps:
+
+1. Providing a field in a record of the table data that indicates whether that row of data is intended to be a parent row. Records that are intended to be parents must set the value of the field, whose name is specified by the `forceExpandCollapseFieldName` attribute, to `true`.
+2. After providing the current data to the `Table` via the `setData` method, all rows that have a value of `true` in the field specified by the `forceExpandCollapseFieldName` attribute will display and expand/collapse button.
+3. Users must register a handler for the `row-expand-toggle` event on the `Table` instance, and will receive that event upon clicking the expand/collapse button.
+4. The details of the handled event will include the id for the row that was expanded.
+5. The user must then create a set of data with rows where they provide a field with the name specified by the `parendIdFieldName` attribute that has the value of the `recordId` value supplied in the event details.
+6. The user then sets the data on the `Table` again with the `setData` method.
+
+_Note: It is up to the user to manage whether or not the children of a row has already been loaded in order to avoid recreation of their data and calling `setData` on the `Table` unnecessarily._
```

Comment:
```suggestion
_The client is responsible for checking if the row’s children have already been loaded. This can prevent unnecessary data recreation and `setData` calls on the `Table`.._
```